### DISCUSSION TITLE
tutorials/lift-off-part1/07-the-frontend-app

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#17] [course feedback]
# tutorials/lift-off-part1/07-the-frontend-app

https://www.apollographql.com/tutorials/lift-off-part1/07-the-frontend-app

**Comment by luis2906** [comment]
I love this tutorial, it is pretty good, thanks so much.

**Reply by R4ph-t** [reply]
We're glad you're enjoying it! Thanks for learning with us ;)

**Reply by abzResponsible** [reply]
I believe this to be a well thought out course!

**Comment by CMH7** [comment]
I only got a little knowledge and experience on React.js but I think I can follow this up. PS. Currently I am using SvelteKit but later its on my list to learn React and Next <3

**Comment by nccharles250** [comment]
I like it

**Comment by le-el** [comment]
I appreciate your course.
Great persons

**Comment by repiemm** [comment]
This part was pretty fast

**Comment by natalieacevedo** [comment]
Amazing tutorial, thanks so much. Documentation is seldom this well explained, kudos!

**Comment by NishantCoder108** [comment]
The Frontend App Done 🚀

**Comment by azimaro** [comment]
I am getting the following error when I try to start from the client directory:
'''> catstronauts-client@1.0.0 start
> vite


  VITE v4.5.2  ready in 344 ms

  ➜  Local:   http://localhost:3000/
  ➜  press h to show help
✘ [ERROR] Unterminated string literal

    node_modules/@react-aria/utils/dist/main.js.map:1:82944:
      1 │ ...etElement.getBoundingClientRect();\n\n      // use scrollIntoView({block: 'nearest'}) instead of .focus to check if the element is fully in view or not since .focus()\n      // won't cause a scroll if the elemen
        ╵    '''

**Reply by azimaro** [reply]
Ok, this is the full error:
client % npm start

> catstronauts-client@1.0.0 start
> vite


  VITE v4.5.2  ready in 344 ms

  ➜  Local:   http://localhost:3000/
  ➜  press h to show help
✘ [ERROR] Unterminated string literal

    node_modules/@react-aria/utils/dist/main.js.map:1:82944:
      1 │ ...etElement.getBoundingClientRect();\n\n      // use scrollIntoView({block: 'nearest'}) instead of .focus to check if the element is fully in view or not since .focus()\n      // won't cause a scroll if the elemen
        ╵                                                                                                                                                                                                                       ^

  The source map "node_modules/@react-aria/utils/dist/main.js.map" was referenced by the file
  "node_modules/@react-aria/utils/dist/main.js" here:

    node_modules/@react-aria/utils/dist/main.js:1336:21:
      1336 │ //# sourceMappingURL=main.js.map
           ╵                      ~~~~~~~~~~~

/Users/angelinazimaro/Desktop/Castronauts/odyssey-lift-off-part1/client/node_modules/esbuild/lib/main.js:1649
  let error = new Error(`${text}${summary}`);
              ^

Error: Build failed with 1 error:
node_modules/@react-aria/utils/dist/main.js.map:1:82944: ERROR: Unterminated string literal
    at failureErrorWithLog (/Users/angelinazimaro/Desktop/Castronauts/odyssey-lift-off-part1/client/node_modules/esbuild/lib/main.js:1649:15)
    at /Users/angelinazimaro/Desktop/Castronauts/odyssey-lift-off-part1/client/node_modules/esbuild/lib/main.js:1058:25
    at /Users/angelinazimaro/Desktop/Castronauts/odyssey-lift-off-part1/client/node_modules/esbuild/lib/main.js:1525:9
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errors: [
    {
      detail: undefined,
      id: '',
      location: {
        column: 82944,
        file: 'node_modules/@react-aria/utils/dist/main.js.map',
        length: 0,
        line: 1,
        lineText: '{"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA;;;;;;;;;;ACAA;;;;;;;;;;CAUC;ACVD;;;;;;;;;;CAUC;AAOM,MAAM,4CAAkB,OAAO,aAAa,cAC/C,CAAA,GAAA,sCAAI,EAAE,eAAe,GACrB,KAAO;;;;ACnBX;;;;;;;;;;CAUC;ACVD;;;;;;;;;;CAUC;;AAKM,SAAS,0CAAmC,EAAM;IACvD,MAAM,MAAM,CAAA,GAAA,mBAAK,EAAwB;IACzC,CAAA,GAAA,yCAAc,EAAE;QACd,IAAI,OAAO,GAAG;IAChB,GAAG;QAAC;KAAG;IACP,aAAa;IACb,OAAO,CAAA,GAAA,wBAAU,EAAK,CAAC,GAAG;QACxB,MAAM,IAAI,IAAI,OAAO;QACrB,OAAO,cAAA,wBAAA,KAAO;IAChB,GAAG,EAAE;AACP;;;ADJO,SAAS,0CAAkB,YAA2B;IAC3D,IAAI,CAAC,OAAO,SAAS,GAAG,CAAA,GAAA,qBAAO,EAAE;IACjC,IAAI,SAAgD,CAAA,GAAA,mBAAK,EAAuB;IAEhF,0EAA0E;IAC1E,yCAAyC;IACzC,IAAI,UAAU,CAAA,GAAA,yCAAa,EAAE;QAC3B,IAAI,CAAC,OAAO,OAAO,EACjB;QAEF,uCAAuC;QACvC,IAAI,WAAW,OAAO,OAAO,CAAC,IAAI;QAElC,8CAA8C;QAC9C,IAAI,SAAS,IAAI,EAAE;YACjB,OAAO,OAAO,GAAG;YACjB;QACF;QAEA,iDAAiD;QACjD,8CAA8C;QAC9C,8DAA8D;QAC9D,IAAI,UAAU,SAAS,KAAK,EAC1B;aAEA,SAAS,SAAS,KAAK;IAE3B;IAEA,CAAA,GAAA,yCAAc,EAAE;QACd,uEAAuE;QACvE,IAAI,OAAO,OAAO,EAChB;IAEJ;IAEA,IAAI,QAAQ,CAAA,GAAA,yCAAa,EAAE,CAAA;QACzB,OAAO,OAAO,GAAG,GAAG;QACpB;IACF;IAEA,OAAO;QAAC;QAAO;KAAM;AACvB;;;AF9CA,mFAAmF;AACnF,IAAI,kCAAY,QACd,OAAO,WAAW,eAClB,OAAO,QAAQ,IACf,OAAO,QAAQ,CAAC,aAAa;AAG/B,IAAI,sCAAkD,IAAI;AAMnD,SAAS,0CAAM,SAAkB;IACtC,IAAI,CAAC,OAAO,SAAS,GAAG,CAAA,GAAA,qBAAO,EAAE;IACjC,IAAI,SAAS,CAAA,GAAA,mBAAK,EAAE;IAEpB,IAAI,MAAM,CAAA,GAAA,gCAAW,EAAE;IAEvB,IAAI,cAAc,CAAA,GAAA,wBAAU,EAAE,CAAC;QAC7B,OAAO,OAAO,GAAG;IACnB,GAAG,EAAE;IAEL,IAAI,iCACF,oCAAc,GAAG,CAAC,KAAK;IAGzB,CAAA,GAAA,yCAAc,EAAE;QACd,IAAI,IAAI;QACR,OAAO;YACL,oCAAc,MAAM,CAAC;QACvB;IACF,GAAG;QAAC;KAAI;IAER,uEAAuE;IACvE,2BAA2B;IAC3B,CAAA,GAAA,sBAAQ,EAAE;QACR,IAAI,QAAQ,OAAO,OAAO;QAC1B,IAAI,OAAO;YACT,OAAO,OAAO,GAAG;YACjB,SAAS;QACX;IACF;IAEA,OAAO;AACT;AAMO,SAAS,0CAAS,GAAW,EAAE,GAAW;IAC/C,IAAI,QAAQ,KACV,OAAO;IAGT,IAAI,SAAS,oCAAc,GAAG,CAAC;IAC/B,IAAI,QAAQ;QACV,OAAO;QACP,OAAO;IACT;IAEA,IAAI,SAAS,oCAAc,GAAG,CAAC;IAC/B,IAAI,QAAQ;QACV,OAAO;QACP,OAAO;IACT;IAEA,OAAO;AACT;AAOO,SAAS,0CAAU,WAA+B,EAAE;IACzD,IAAI,KAAK;IACT,IAAI,CAAC,YAAY,cAAc,GAAG,CAAA,GAAA,yCAAa,EAAE;IACjD,IAAI,WAAW,CAAA,GAAA,wBAAU,EAAE;QACzB,cAAc;YACZ,MAAM;YAEN,MAAM,SAAS,cAAc,CAAC,MAAM,KAAK;QAC3C;IACF,GAAG;QAAC;QAAI;KAAc;IAEtB,CAAA,GAAA,yCAAc,EAAE,UAAU;QAAC;QAAI;WAAa;KAAS;IAErD,OAAO;AACT;;CDjGC;AKVD;;;;;;;;;;CAUC,GAED;;CAEC,GACM,SAAS,0CAAM,GAAG,SAAgB;IACvC,OAAO,CAAC,GAAG;QACT,KAAK,IAAI,YAAY,UACnB,IAAI,OAAO,aAAa,YACtB,YAAY;IAGlB;AACF;;;ACvBO,MAAM,4CAAmB,CAAC;QACxB;IAAP,OAAO,CAAA,oBAAA,eAAA,yBAAA,GAAI,aAAa,cAAjB,+BAAA,oBAAqB;AAC9B;AAEO,MAAM,4CAAiB,CAC5B;IAEA,IAAI,MAAM,YAAY,MAAM,GAAG,MAAM,KAAK,IACxC,OAAO;IAGT,MAAM,MAAM,0CAAiB;IAC7B,OAAO,IAAI,WAAW,IAAI;AAC5B;;;ACbA;;;;;;;;;;CAUC;;;AAyBM,SAAS,0CAAiC,GAAG,IAAO;IACzD,oFAAoF;IACpF,uDAAuD;IACvD,IAAI,SAAgB;QAAC,GAAG,IAAI,CAAC,EAAE;IAAA;IAC/B,IAAK,IAAI,IAAI,GAAG,IAAI,KAAK,MAAM,EAAE,IAAK;QACpC,IAAI,QAAQ,IAAI,CAAC,EAAE;QACnB,IAAK,IAAI,OAAO,MAAO;YACrB,IAAI,IAAI,MAAM,CAAC,IAAI;YACnB,IAAI,IAAI,KAAK,CAAC,IAAI;YAElB,eAAe;YACf,IACE,OAAO,MAAM,cACb,OAAO,MAAM,cACb,qCAAqC;YACrC,GAAG,CAAC,EAAE,KAAK,OACX,GAAG,CAAC,EAAE,KAAK,OACX,IAAI,UAAU,CAAC,MAAM,OAAO,GAAG,MAC/B,IAAI,UAAU,CAAC,MAAM,OAAO,GAAG,IAE/B,MAAM,CAAC,IAAI,GAAG,CAAA,GAAA,yCAAI,EAAE,GAAG;iBAGlB,IACL,AAAC,CAAA,QAAQ,eAAe,QAAQ,kBAAiB,KACjD,OAAO,MAAM,YACb,OAAO,MAAM,UAEb,MAAM,CAAC,IAAI,GAAG,CAAA,GAAA,qCAAG,EAAE,GAAG;iBACjB,IAAI,QAAQ,QAAQ,KAAK,GAC9B,OAAO,EAAE,GAAG,CAAA,GAAA,yCAAO,EAAE,GAAG;iBAGxB,MAAM,CAAC,IAAI,GAAG,MAAM,YAAY,IAAI;QAExC;IACF;IAEA,OAAO;AACT;;;AC1EA;;;;;;;;;;CAUC,GAOM,SAAS,0CAAa,GAAG,IAAuB;IACrD,IAAI,KAAK,MAAM,KAAK,GAClB,OAAO,IAAI,CAAC,EAAE;IAGhB,OAAO,CAAC;QACN,KAAK,IAAI,OAAO,KAAM;YACpB,IAAI,OAAO,QAAQ,YACjB,IAAI;iBACC,IAAI,OAAO,MAChB,IAAI,OAAO,GAAG;QAElB;IACF;AACF;;;AC/BA;;;;;;;;;;CAUC,GAID,MAAM,qCAAe,IAAI,IAAI;IAC3B;CACD;AAED,MAAM,2CAAqB,IAAI,IAAI;IACjC;IACA;IACA;IACA;CACD;AAED,gCAAgC;AAChC,MAAM,sCAAgB,IAAI,IAAI;IAC5B;IACA;IACA;IACA;IACA;IACA;CACD;AAeD,MAAM,+BAAS;AAOR,SAAS,0CAAe,KAAkD,EAAE,OAAgB,CAAC,CAAC;IACnG,IAAI,aAAC,SAAS,UAAE,MAAM,aAAE,SAAS,EAAC,GAAG;IACrC,IAAI,gBAAgB,CAAC;IAErB,IAAK,MAAM,QAAQ,MACjB,IACE,OAAO,SAAS,CAAC,cAAc,CAAC,IAAI,CAAC,OAAO,SAC1C,CAAA,mCAAa,GAAG,CAAC,SAChB,aAAa,yCAAmB,GAAG,CAAC,SACpC,UAAU,oCAAc,GAAG,CAAC,UAC7B,sBAAA,gCAAA,UAAW,GAAG,CAAC,UACf,6BAAO,IAAI,CAAC,KAAI,GAGlB,aAAa,CAAC,KAAK,GAAG,KAAK,CAAC,KAAK;IAIrC,OAAO;AACT;;;AC1EA;;;;;;;;;;CAUC,GAoBM,SAAS,0CAAsB,OAAyB;IAC7D,IAAI,+CACF,QAAQ,KAAK,CAAC;QAAC,eAAe;IAAI;SAC7B;QACL,IAAI,qBAAqB,4CAAsB;QAC/C,QAAQ,KAAK;QACb,4CAAsB;IACxB;AACF;AAEA,IAAI,oDAA8C;AAClD,SAAS;IACP,IAAI,qDAA+B,MAAM;QACvC,oDAA8B;QAC9B,IAAI;YACF,IAAI,YAAY,SAAS,aAAa,CAAC;YACvC,UAAU,KAAK,CAAC;gBACd,IAAI,iBAAgB;oBAClB,oDAA8B;oBAC9B,OAAO;gBACT;YACF;QACF,EAAE,OAAO,GAAG;QACV,SAAS;QACX;IACF;IAEA,OAAO;AACT;AAEA,SAAS,4CAAsB,OAAyB;IACtD,IAAI,SAAS,QAAQ,UAAU;IAC/B,IAAI,qBAA0C,EAAE;IAChD,IAAI,uBAAuB,SAAS,gBAAgB,IAAI,SAAS,eAAe;IAEhF,MAAO,kBAAkB,eAAe,WAAW,qBAAsB;QACvE,IACE,OAAO,YAAY,GAAG,OAAO,YAAY,IACzC,OAAO,WAAW,GAAG,OAAO,WAAW,EAEvC,mBAAmB,IAAI,CAAC;YACtB,SAAS;YACT,WAAW,OAAO,SAAS;YAC3B,YAAY,OAAO,UAAU;QAC/B;QAEF,SAAS,OAAO,UAAU;IAC5B;IAEA,IAAI,gCAAgC,aAClC,mBAAmB,IAAI,CAAC;QACtB,SAAS;QACT,WAAW,qBAAqB,SAAS;QACzC,YAAY,qBAAqB,UAAU;IAC7C;IAGF,OAAO;AACT;AAEA,SAAS,4CAAsB,kBAAuC;IACpE,KAAK,IAAI,WAAC,OAAO,aAAE,SAAS,cAAE,UAAU,EAAC,IAAI,mBAAoB;QAC/D,QAAQ,SAAS,GAAG;QACpB,QAAQ,UAAU,GAAG;IACvB;AACF;;;AC/FA;;;;;;;;;;CAUC,GAEM,SAAS,0CAAU,OAAO,EAAE,OAAO,EAAE,cAAc,YAAY;IACpE,IAAI,OAAO,QAAQ,qBAAqB;IACxC,IAAI,SACF,OAAO,gBAAgB,eAAe,KAAK,KAAK,GAAG,KAAK,MAAM;IAEhE,OAAO,gBAAgB,eAAe,KAAK,IAAI,GAAG,KAAK,GAAG;AAC5D;;;AClBA;;;;;;;;;;ACAA;;;;;;;;;;CAUC,GAED,SAAS,oCAAc,EAAU;QAK7B;IAJF,IAAI,OAAO,WAAW,eAAe,OAAO,SAAS,IAAI,MACvD,OAAO;IAET,OAAO,EACL,kCAAA,OAAO,SAAS,CAAC,gBAAgB,cAAjC,sDAAA,gCAAmC,MAAM,CAAC,IAAI,CAAC,CAAC,QAA4C,GAAG,IAAI,CAAC,MAAM,KAAK,OAEjH,GAAG,IAAI,CAAC,OAAO,SAAS,CAAC,SAAS;AACpC;AAEA,SAAS,mCAAa,EAAU;QAElB;IADZ,OAAO,OAAO,WAAW,eAAe,OAAO,SAAS,IAAI,OACxD,GAAG,IAAI,CAAC,EAAA,kCAAA,OAAO,SAAS,CAAC,gBAAgB,cAAjC,sDAAA,gCAAmC,QAAQ,KAAI,OAAO,SAAS,CAAC,QAAQ,IAChF;AACN;AAEO,SAAS;IACd,OAAO,mCAAa;AACtB;AAEO,SAAS;IACd,OAAO,mCAAa;AACtB;AAEO,SAAS;IACd,OAAO,mCAAa,aAClB,yFAAyF;IACxF,+CAAW,UAAU,cAAc,GAAG;AAC3C;AAEO,SAAS;IACd,OAAO,8CAAc;AACvB;AAEO,SAAS;IACd,OAAO,+CAAW;AACpB;AAEO,SAAS;IACd,OAAO,oCAAc,mBAAmB,CAAC;AAC3C;AAEO,SAAS;IACd,OAAO,oCAAc;AACvB;AAEO,SAAS;IACd,OAAO,oCAAc;AACvB;AAEO,SAAS;IACd,OAAO,oCAAc;AACvB;;CDtDC;;;AAYD,MAAM,oDAAgB,CAAA,GAAA,0BAAY,EAAU;IAC1C,UAAU;IACV,MAAM;AACR;AAWO,SAAS,0CAAe,KAA0B;IACvD,IAAI,YAAC,QAAQ,YAAE,QAAQ,EAAC,GAAG;IAE3B,IAAI,MAAM,CAAA,GAAA,oBAAM,EAAE,IAAO,CAAA;YACvB,UAAU;YACV,MAAM,CAAC,QAAiB;gBACtB,uCAAiB,QAAQ,CAAA;oBACvB,IAAI,0CAAqB,MAAM,YAC7B,SAAS,KAAK,QAAQ,GAAG,KAAK,MAAM,GAAG,KAAK,IAAI;yBAEhD,0CAAS,MAAM;gBAEnB;YACF;QACF,CAAA,GAAI;QAAC;KAAS;IAEd,qBACE,0DAAC,oCAAc,QAAQ;QAAC,OAAO;OAC5B;AAGP;AAEO,SAAS;IACd,OAAO,CAAA,GAAA,uBAAS,EAAE;AACpB;AASO,SAAS,0CAAqB,IAAuB,EAAE,SAAoB;IAChF,qHAAqH;IACrH,IAAI,SAAS,KAAK,YAAY,CAAC;IAC/B,OACE,AAAC,CAAA,CAAC,UAAU,WAAW,OAAM,KAC7B,KAAK,MAAM,KAAK,SAAS,MAAM,IAC/B,CAAC,KAAK,YAAY,CAAC,eACnB,CAAC,UAAU,OAAO,IAAI,wBAAwB;IAC9C,CAAC,UAAU,OAAO,IAAI,4BAA4B;IAClD,CAAC,UAAU,MAAM,IAAI,WAAW;IAChC,CAAC,UAAU,QAAQ;AAEvB;AAEO,SAAS,0CAAS,MAAyB,EAAE,SAAoB,EAAE,aAAa,IAAI;QAOtE,oBAAA;IANnB,IAAI,WAAC,OAAO,WAAE,OAAO,UAAE,MAAM,YAAE,QAAQ,EAAC,GAAG;IAE3C,gGAAgG;IAChG,gGAAgG;IAChG,6GAA6G;IAC7G,mHAAmH;IACnH,IAAI,CAAA,GAAA,yCAAQ,SAAO,gBAAA,OAAO,KAAK,cAAZ,qCAAA,qBAAA,cAAc,IAAI,cAAlB,yCAAA,mBAAoB,UAAU,CAAC,WAAU,OAAO,MAAM,KAAK;QAC5E,IAAI,CAAA,GAAA,yCAAI,KACN,UAAU;aAEV,UAAU;;IAId,oGAAoG;IACpG,gIAAgI;IAChI,IAAI,QAAQ,CAAA,GAAA,yCAAO,OAAO,CAAA,GAAA,yCAAI,OAAO,CAAC,CAAA,GAAA,yCAAK,OAAO,OAE9C,IAAI,cAAc,WAAW;QAAC,eAAe;iBAAS;iBAAS;gBAAS;kBAAQ;IAAQ,KACxF,IAAI,WAAW,SAAS;iBAAC;iBAAS;gBAAS;kBAAQ;QAAU,SAAS;QAAM,YAAY;IAAI;IAC/F,0CAAiB,SAAS,GAAG;IAC9B,CAAA,GAAA,yCAAoB,EAAE;IACtB,OAAO,aAAa,CAAC;IACpB,0CAAiB,SAAS,GAAG;AAChC;AACA,uDAAuD;AACtD,0CAAiB,SAAS,GAAG;AAE9B,SAAS,uCAAiB,MAAe,EAAE,IAAuC;IAChF,IAAI,kBAAkB,mBACpB,KAAK;SACA,IAAI,OAAO,YAAY,CAAC,cAAc;QAC3C,IAAI,OAAO,SAAS,aAAa,CAAC;QAClC,KAAK,IAAI,GAAG,OAAO,YAAY,CAAC;QAChC,IAAI,OAAO,YAAY,CAAC,gBACtB,KAAK,MAAM,GAAG,OAAO,YAAY,CAAC;QAEpC,IAAI,OAAO,YAAY,CAAC,aACtB,KAAK,GAAG,GAAG,OAAO,YAAY,CAAC;QAEjC,IAAI,OAAO,YAAY,CAAC,kBACtB,KAAK,QAAQ,GAAG,OAAO,YAAY,CAAC;QAEtC,IAAI,OAAO,YAAY,CAAC,cACtB,KAAK,IAAI,GAAG,OAAO,YAAY,CAAC;QAElC,IAAI,OAAO,YAAY,CAAC,yBACtB,KAAK,cAAc,GAAG,OAAO,YAAY,CAAC;QAE5C,OAAO,WAAW,CAAC;QACnB,KAAK;QACL,OAAO,WAAW,CAAC;IACrB;AACF;AAEA,SAAS,wCAAkB,MAAe,EAAE,SAAoB;IAC9D,uCAAiB,QAAQ,CAAA,OAAQ,0CAAS,MAAM;AAClD;AAEO,SAAS,0CAAsB,KAAmB;IACvD,OAAO;QACL,aAAa,MAAM,IAAI;QACvB,eAAe,MAAM,MAAM;QAC3B,YAAY,MAAM,GAAG;QACrB,iBAAiB,MAAM,QAAQ;QAC/B,aAAa,MAAM,IAAI;QACvB,wBAAwB,MAAM,cAAc;IAC9C;AACF;;;AEzJA;;;;;;;;;;CAUC,GAED,uEAAuE;AACvE,6EAA6E;AAC7E,iFAAiF;AACjF,mFAAmF;AACnF,mFAAmF;AACnF,wCAAwC;AACxC,IAAI,6CAAuB,IAAI;AAE/B,wEAAwE;AACxE,IAAI,4CAAsB,IAAI;AAE9B,SAAS;IACP,IAAI,OAAO,WAAW,aACpB;IAGF,SAAS,kBAAkB,KAAY;QACrC,OAAO,kBAAkB;IAC3B;IAEA,IAAI,oBAAoB,CAAC;QACvB,IAAI,CAAC,kBAAkB,MAAM,CAAC,EAAE,MAAM,EACpC;QAEF,+DAA+D;QAC/D,IAAI,cAAc,2CAAqB,GAAG,CAAC,EAAE,MAAM;QACnD,IAAI,CAAC,aAAa;YAChB,cAAc,IAAI;YAClB,2CAAqB,GAAG,CAAC,EAAE,MAAM,EAAE;YAEnC,+FAA+F;YAC/F,yGAAyG;YACzG,mGAAmG;YACnG,EAAE,MAAM,CAAC,gBAAgB,CAAC,oBAAoB,iBAAiB;gBAC7D,MAAM;YACR;QACF;QAEA,YAAY,GAAG,CAAC,EAAE,YAAY;IAChC;IAEA,IAAI,kBAAkB,CAAC;QACrB,IAAI,CAAC,kBAAkB,MAAM,CAAC,EAAE,MAAM,EACpC;QAEF,yDAAyD;QACzD,IAAI,aAAa,2CAAqB,GAAG,CAAC,EAAE,MAAM;QAClD,IAAI,CAAC,YACH;QAGF,WAAW,MAAM,CAAC,EAAE,YAAY;QAEhC,2GAA2G;QAC3G,IAAI,WAAW,IAAI,KAAK,GAAG;YACzB,EAAE,MAAM,CAAC,mBAAmB,CAAC,oBAAoB;YACjD,2CAAqB,MAAM,CAAC,EAAE,MAAM;QACtC;QAEA,kEAAkE;QAClE,IAAI,2CAAqB,IAAI,KAAK,GAAG;YACnC,KAAK,IAAI,MAAM,0CACb;YAGF,0CAAoB,KAAK;QAC3B;IACF;IAEA,SAAS,IAAI,CAAC,gBAAgB,CAAC,iBAAiB;IAChD,SAAS,IAAI,CAAC,gBAAgB,CAAC,iBAAiB;AAClD;AAEA,IAAI,OAAO,aAAa;IACtB,IAAI,SAAS,UAAU,KAAK,WAC1B;SAEA,SAAS,gBAAgB,CAAC,oBAAoB;;AAI3C,SAAS,0CAAmB,EAAc;IAC/C,4EAA4E;IAC5E,sBAAsB;QACpB,gEAAgE;QAChE,+EAA+E;QAC/E,IAAI,2CAAqB,IAAI,KAAK,GAChC;aAEA,0CAAoB,GAAG,CAAC;IAE5B;AACF;;;ACxGA;;;;;;;;;;CAUC,GAEA,uCAAuC;;AAoBxC,oFAAoF;AACpF,sFAAsF;AACtF,oFAAoF;AACpF,MAAM,yCAAkC,EAAE;AAOnC,SAAS,0CAAU,KAAqB;IAC7C,QAAQ,IAAI,CAAC;IACb,IAAI,gBAAC,YAAY,WAAE,OAAO,eAAE,WAAW,WAAE,OAAO,UAAE,MAAM,oBAAE,gBAAgB,eAAE,WAAW,eAAE,W'... 72942 more characters,
        namespace: '',
        suggestion: ''
      },
      notes: [
        {
          location: {
            column: 21,
            file: 'node_modules/@react-aria/utils/dist/main.js',
            length: 11,
            line: 1336,
            lineText: '//# sourceMappingURL=main.js.map',
            namespace: '',
            suggestion: ''
          },
          text: 'The source map "node_modules/@react-aria/utils/dist/main.js.map" was referenced by the file "node_modules/@react-aria/utils/dist/main.js" here:'
        }
      ],
      pluginName: '',
      text: 'Unterminated string literal'
    }
  ],
  warnings: []
}

**Reply by azimaro** [reply]
localhost:3000 is blank

**Reply by lizhennessy** [reply]
Hi @azimaro - I'm happy to take a look. Are you on the JavaScript or the TypeScript version of the course? Have you tried deleting the `node_modules` folder and re-running `npm install` in the `client` directory? Also, curious what version of Node you're running - you can find out by running `node -v` in the terminal.

**Reply by azimaro** [reply]
I am on the Typescript version, and I have not tried that, but I will. Thanks!

**Reply by azimaro** [reply]
node 20.11.1

**Reply by azimaro** [reply]
Deleting the node_modules folder and re-running npm install in the client dir worked! (Although the install got a lot of deprecated warnings)/ I can now see the frame for Castronauts in localhost:3000 
Thank you!

**Reply by lizhennessy** [reply]
Oh I'm so glad to hear that @azimaro ! I wonder if something went a little strange with the first installation. Happy that it's working for you now. 😊

**Comment by HNBOLLA** [comment]
I'm having trouble loading localhost when start client and is opened at http://localhost:3001/ . Its in loading state for longer and  no output. I'm on Typescript version of tutorial. Its running on node v22.13.1 and tried to delete node_modules and reinstall , but still same. Any help is appreciated.
➜ npm start

> catstronauts-client@1.0.0 start
> vite


  VITE v4.5.9  ready in 225 ms

  ➜  Local:   http://localhost:3000/
  ➜  press h to show help

**Reply by mabuyo** [reply]
Hi there @HNBOLLA - sorry to hear you're having issues. I can't reproduce your issue with the same node version unfortunately. Can you clarify what you mean by "loading state"? And is there no output in the browser console, or the terminal console where you ran `npm start`? It would also help to see what you have in `index.tsx`. Hope we can troubleshoot this together!

**Reply by HNBOLLA** [reply]
Hi Mabuyo, Thanks for helping. I mean, there is no output in the browser console, and below are the index.tsx contents.
import React from 'react';
import { createRoot } from 'react-dom/client'
import GlobalStyles from './styles';
import Pages from './pages';

const root = createRoot(document.getElementById('root')!);

root.render(
  <React.StrictMode>
    <GlobalStyles />
    <Pages />
  </React.StrictMode>
);

**Reply by mabuyo** [reply]
@HNBOLLA It looks like you're missing some code within the `StrictMode` tags. The starting point for the front-end app should be what's in this file: https://github.com/apollographql/odyssey-lift-off-part1/blob/typescript/client/src/index.tsx

**Reply by HNBOLLA** [reply]
@mabuyo, no those lines of code are present in the file, but for some reason, those two lines are showing blank in my reply. They are not missing.

**Reply by mabuyo** [reply]
OK got it! So if you have the correct starting point as [indicated in this file](https://github.com/apollographql/odyssey-lift-off-part1/blob/typescript/client/src/index.tsx), by the end of this lesson, you should be seeing an empty layout with a a header and footer:

![image](https://github.com/user-attachments/assets/025a563f-4b26-4b72-a571-cbaf03f5b364)

Can you clarify what you are seeing instead?

**Reply by HNBOLLA** [reply]
ayyyyy.. now I see the empty layout as above with logo. There aren't any changes but somehow it worked now. Thanks for your time and help @Mabuyo.

**Reply by mabuyo** [reply]
Oh I'm glad to hear it! Enjoy the rest of the course and let us know if you run into any other issues :)

**Comment by nishantpdev** [comment]
facing issue with '@emotion/core'? (@emotion/core,is deprecated and has been replaced by @emotion/react since Emotion v11.)
do this:
1. Uninstall @emotion/core
Run this to remove it: `npm uninstall @emotion/core`
2. Verify it's gone
Check it’s no longer in your package.json: `npm ls @emotion/core`
It should now only show it as a transitive dependency (i.e. used by some other package like @apollo/space-kit), or not at all.
3. Now Add the Override
Update your package.json with:
`"overrides": {
  "@emotion/core": "npm:@emotion/react@^11.0.0"
}`
4. Reinstall Dependencies
Now clear and reinstall everything:
`rm -rf node_modules package-lock.json`
`npm install`
If you're on windows and having issue with `rm -rf ...` command: manually delete `node_modules` folder and `package-lock.json` file.
5. Final Sanity Check
If all goes well, you should now have:
 - No direct dependency on @emotion/core
 - Any internal usage of it (e.g. from @apollo/space-kit) now redirected to @emotion/react :  `npm ls @emotion/core`
6. Lastly run : `npm start`

Happy coding! 🦕

**Comment by alison-hernandez** [comment]
I had an issue where `npm start` rendered a blank page, with the following error in the console: Uncaught SyntaxError: The requested module '/src/styles.js' does not provide an export named 'default' (at index.tsx:3:8). 

To fix, I needed to delete a bunch of files that were "nested" in styles.tsx (stuff like styles.js, styles.d.ts, index.js, etc). Not sure how those old compile files got in there, but once I deleted them the page instantly refreshed and I could see the header and footer correctly. Hopefully this helps anyone else who runs into this!
